frontend/src/main/utils/commons.js

```diff
@@ -0,0 +1,17 @@
+import { useQuery } from "react-query";
+import axios from "axios";
+
+export function useCommons() {
+  return useQuery("commons", async () => {
+    const uri = "/api/commons";
+    try {
+      const response = await axios.get(uri);      
+      return response.data ;
+    } catch (e) {
+      console.error(`Error getting data from ${uri}:`,e);
```

Comment:
I wonder if we should put up an error toast... so that users are aware that something is going awry?   We don't necessarily have to do that in this PR, or hold up merging this for that.. I'm just noting it.